core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -55,7 +55,11 @@ public class RESTCatalog
   public RESTCatalog() {
     this(
         SessionCatalog.SessionContext.createEmpty(),
-        config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build());
+        config ->
```

Comment:
we haven't been setting the headers here previously which might be because it was missed earlier. I believe it should be fine to set them here

Reply:
can you please add a test to `TestRESTCatalog` that make sure that headers are properly passed from the `properties` to the underlying HTTP client when a request is being done?